openbas-front/src/components/common/queryable/useQueryableWithLocalStorage.tsx

```diff
@@ -70,9 +72,21 @@ export const useQueryable = (initSearchPaginationInput: Partial<SearchPagination
 };
 
 export const useQueryableWithLocalStorage = (localStorageKey: string, initSearchPaginationInput: Partial<SearchPaginationInput>) => {
+  const [searchParams] = useSearchParams();
   const finalSearchPaginationInput: SearchPaginationInput = buildSearchPagination(initSearchPaginationInput);
+  const searchPaginationInputFromUri = retrieveFromUri(localStorageKey, searchParams);
 
-  const [searchPaginationInput, setSearchPaginationInput] = useLocalStorage<SearchPaginationInput>(localStorageKey, finalSearchPaginationInput);
+  const [searchPaginationInputFromLocalStorage, setSearchPaginationInputFromLocalStorage] = useLocalStorage<SearchPaginationInput>(
+    localStorageKey,
+    searchPaginationInputFromUri ?? finalSearchPaginationInput,
+  );
+  const [searchPaginationInput, setSearchPaginationInput] = useState(searchPaginationInputFromUri ?? searchPaginationInputFromLocalStorage);
```

Comment:
```suggestion
  // add a transitional state to avoid re render because of useLocalStorage hook
  const [searchPaginationInput, setSearchPaginationInput] = useState(searchPaginationInputFromUri ?? searchPaginationInputFromLocalStorage);
```